test-junit5/src/main/java/io/micronaut/test/extensions/junit5/MicronautJunit5Extension.java

```diff
@@ -213,16 +221,42 @@ private TestContext buildContext(ExtensionContext context) {
 
     @Override
     public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-        return applicationContext != null && applicationContext.containsBean(parameterContext.getParameter().getType());
+        final Argument<?> argument = getArgument(parameterContext, applicationContext);
+        if (argument != null) {
+            if (argument.isAnnotationPresent(Value.class) || argument.isAnnotationPresent(Property.class)) {
+                return true;
+            } else {
+                return applicationContext.containsBean(argument.getType(), resolveQualifier(argument));
+            }
+        }
+        return false;
     }
 
     @Override
     public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-        Named named = parameterContext.findAnnotation(Named.class).orElse(null);
-        if (named != null) {
-            return applicationContext.getBean(parameterContext.getParameter().getType(), Qualifiers.byName(named.value()));
+        final Argument<?> argument = getArgument(parameterContext, applicationContext);
+        if (argument != null) {
+            Optional<String> v = argument.getAnnotationMetadata().stringValue(Value.class);
+            if (v.isPresent()) {
+                Optional<String> finalV = v;
+                return applicationContext.getEnvironment().getProperty(v.get(), argument)
+                        .orElseThrow(() ->
+                    new ParameterResolutionException("Unresolvable property specified to @Value: " + finalV.get())
+                );
+            } else {
+                v = argument.getAnnotationMetadata().stringValue(Property.class, "name");
+                if (v.isPresent()) {
+                    Optional<String> finalV1 = v;
+                    return applicationContext.getEnvironment()
+                            .getProperty(v.get(), argument).orElseThrow(() ->
+                                    new ParameterResolutionException("Unresolvable property specified to @Property: " + finalV1.get())
+                            );
+                } else {
+                    return applicationContext.getBean(argument.getType(), resolveQualifier(argument));
```

Comment:
Note when Micronaut 3.0 is out there should be just argument and not argument.getType() to support generic types